pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -267,6 +267,12 @@ func (p Provider) DestroyBootstrap(ctx context.Context, in clusterapi.BootstrapD
 		return fmt.Errorf("failed to remove bootstrap firewall rules: %w", err)
 	}
 
+	// Remove the overly permissive firewall rules created by CAPG that are redundant with those created by installer
+	logrus.Infof("Removing firewall rules created by cluster-api-provider-gcp")
+	if err := removeCAPGFirewallRules(ctx, in.Metadata.InfraID, projectID); err != nil {
```

Comment:
Hm it looks like this is failing in the gcp xpn e2e test:

> level=fatal msg=error destroying bootstrap resources failed during the destroy bootstrap hook: failed to remove firewall rules created by cluster-api-provider-gcp: failed to delete allow-ci-op-kd627brl-1ea87-f2m9p-cluster firewall rule: googleapi: Error 404: The resource 'projects/openshift-dev-installer/global/firewalls/allow-ci-op-kd627brl-1ea87-f2m9p-cluster' was not found, notFound

It looks like @barbacbd added to capg that it will skip firewall creation for xpn installs:
https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/3b70681f9f9d67ef812e48e65d6cf9e880defb6e/cloud/services/compute/firewalls/reconcile.go#L31-L34

 Is that correct? If so, we should probably just skip the destroy, rather than setting project id.

Reply:
Updated to skip the removal on shared VPC installs